### ISSUE TITLE
Support reading partitioned parquet datasets through _metadata files

### ISSUE THREAD

**Issue by patrikkj** [pola-rs/polars#7707]
### Problem description

It would be great to support parsing `_metadata` and `_custom_metadata` files as implemented in Spark/Pyarrow/Dask 🌟

As described in the [PyArrow docs](https://arrow.apache.org/docs/python/parquet.html#writing-metadata-and-common-metadata-files):

> Those files include information about the schema of the full dataset (for _common_metadata) and potentially all row group metadata of all files in the partitioned dataset as well (for _metadata). The actual files are metadata-only Parquet files. Note this is not a Parquet standard, but a convention set in practice by those frameworks.
>
> Using those files can give a more efficient creation of a parquet Dataset, since it can use the stored schema and and file paths of all row groups, instead of inferring the schema and crawling the directories for all Parquet files (this is especially the case for filesystems where accessing files is expensive).

Despite not being a part of the official parquet standard, supporting for this convention is essential for interop. with systems where this is the agreed upon parquet storage spec. It offers a clear separation between data and metadata and can result in significant performance improvements for remote filesystems.

In `pyarrow` this is implemented in `arrow::datasets::ParquetDatasetFactory` (Link to [Docs](https://arrow.apache.org/docs/cpp/api/dataset.html#_CPPv4N5arrow7dataset21ParquetDatasetFactoryE) / [GitHub](https://github.com/apache/arrow/blob/532b9a57cab4fa1f88438fd0bd79cb7eb8aa2df3/python/pyarrow/_dataset_parquet.pyx#L837)).

Fully working example using `pyarrow`:
```python
import pyarrow as pa
import pyarrow.dataset as ds
import pyarrow.parquet as pq


# Create some example data
table = pa.table({'n_legs': [2, 2, 4, 4, 5, 100],
                  'animal': ["Flamingo", "Parrot", "Dog", "Horse",
                             "Brittle stars", "Centipede"]})

# Write dataset
metadata_collector = []
pq.write_to_dataset(table=table, 
                    root_path="./dummy", 
                    metadata_collector=metadata_collector)

# Write metadata
pq.write_metadata(schema=table.schema, 
                  where="./dummy/_metadata", 
                  metadata_collector=metadata_collector)

# Results in the following file tree:
# .
# └── dummy
#     ├── 4895d51acbc3460bb16f1d058ec89e20-0.parquet
#     └── _metadata

# Read data from metadata file
dataset = ds.parquet_dataset("./dummy/_metadata")

# >>> print(dataset.to_table())
# pyarrow.Table
# n_legs: int64
# animal: string
# ----
# n_legs: [[2,2,4,4,5,100]]
# animal: [["Flamingo","Parrot","Dog","Horse","Brittle stars","Centipede"]]
```

Relevant docs:
- https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_to_dataset.html
- https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_metadata.html
- https://arrow.apache.org/docs/python/generated/pyarrow.dataset.parquet_dataset.html

**Comment by deanm0000** [comment]
Is this meant to nudge the development of native polars datasets and for those to include metadata files? I ask that because polars doesn't do datasets anyways.

**Comment by patrikkj** [comment]
Yup! There is a non-native workaround which is to wrap it as a pyarrow dataset and using `pl.scan_pyarrow_dataset(...)`.

Is there active development to implement native polars datasets? 
By native I mean something along the lines of `pl.scan_dataset(...)`, could not find any related PRs/branches.
Closest would probably be supporting partition-aware parquet scanning, #426.
